### ISSUE TITLE
Allow for centered derivatives

### ISSUE THREAD

**Issue by ryantibs** [cmu-delphi/epiprocess#33]
I'm a bit unsatisfied with the derivative functionality because it is really *only* possible to estimate derivatives using a trailing window, when using smoothing splines or trend filtering. This is because `estimate_deriv()` returns the estimated derivative at the end of the sequence. So even if `align = "center"` is set in the call to `epi_slide()`, it still wouldn't compute the derivative in the middle of the window (unless the `method = "lin"`, where it uses linear regression, and effectively estimates a constant derivative across the entire window).

Need to figure out some way to allow for computing centered derivatives in a way that's not so clunky. E.g., I wouldn't want to have to set both `align = "center"` in `epi_slide()` and then some other argument in `estimate_deriv()` as well. 

While we're at it (revisiting derivatives), I would want to improve the derivatives vignette. Right now it may not be very interpretable by an epi-minded person ... we're just computing and plotting derivatives. Maybe we should normalize by the incidence curve itself and think of these as estimates of growth rates (more sophisticated than what percentage change is giving you).

**Comment by ryantibs** [comment]
Continuing recording thoughts. I think it would be natural to allow for something like this.

```
x %>% epi_slide(deriv = estimate_deriv(x = time_value, y = cases, x0 = Middle(time_value)),
                              n = 28, align = "center")
```

In other words, `estimate_deriv()` takes `x`, `y`, and `x0`, which are the design points and response for the function estimation, and the point at which you want the derivative estimated, respectively.

- If `x` is missing it defaults to evenly-spaced design points across the given response `y`. 
- If `x0` is missing it defaults to `Middle(x)`, where `Middle()` is just a convenience function we prove that takes the middle element of a vector (rounding appropriately if the vector has odd length).

Thus the above is equivalent to 
```
x %>% epi_slide(deriv = estimate_deriv(cases), n = 28, align = "center")
```

And if we really wanted to compute the derivative at the end of the sequence with right-aligned windows, e.g., 
```
x %>% epi_slide(deriv = estimate_deriv(y = cases, x0 = End(time_value)), n = 28, align = "right")
```

**Comment by ryantibs** [comment]
Closed indirectly via #40, where I implemented growth rates in favor of estimating explicit derivatives (and the growth rates are based on centered windows, for any of the local methods).
